osu.Game/Rulesets/Mods/IApplicableToDifficulty.cs

```diff
@@ -15,5 +15,12 @@ public interface IApplicableToDifficulty : IApplicableMod
         /// </summary>
         /// <param name="difficulty">The difficulty to mutate.</param>
         void ApplyToDifficulty(BeatmapDifficulty difficulty);
+
+        /// <summary>
+        /// Whether this mod changes some aspect of a <see cref="BeatmapDifficulty"/>.
+        /// </summary>
+        /// <param name="difficulty">The difficulty to check against.</param>
+        /// <returns></returns>
+        bool ChangesDifficultyAspect(BeatmapDifficulty difficulty);
```

Comment:
This shouldn't be declared on `IApplicableToDifficulty`; it should be declared on `Mod`, and probably have the shape of `bool IsRedundant(IBeatmap)` or similar.

Case in point: we will likely want to apply this to `OsuClassicMod`, because I'm not sure that mod actually does anything once you uncheck all of the checkboxes on it.

Another case in point, which you even bring up: mania key mods (that are not `IApplicableToDifficulty`).

Reply:
I was going to implement a different method for mania key mods, but I think your solution is just better. I will rewrite my implementation and force-push later.